File: implementation/boj26150.cpp
/*
BOJ26150 : Identify, Sort, Index, Solve (S5)

해결 방법 : 전부 입력 받고, sorting 후, 각 단어에서 인덱스 - 1의 문자를 꺼내기
*/

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <cctype>

using namespace std;

struct Info {
  int num;
  string word;
  int rank;
};

int main() {
  int n;
  cin >> n;
  vector<Info> arr;
  for (int i = 0; i < n; i++) {
    string word;
    int num, rank;
    cin >> word >> num >> rank;
    arr.push_back({num, word, rank});
  }
  sort(arr.begin(), arr.end(), [](const Info& a, const Info& b) {
    return a.num < b.num;
  });
  string ans = "";
  for (int i = 0; i < n; i++) {
    int rank = arr[i].rank;
    char temp = arr[i].word[rank-1];
    temp = toupper(temp);
    ans += temp;
  }
  cout << ans << '\n';
};

File: implementation/boj26150.py
'''
BOJ26150 : Identify, Sort, Index, Solve (S5)

해결 방법 : 전부 입력 받고, sorting 후, 각 단어에서 인덱스 - 1의 문자를 꺼내기
'''

n = int(input())
arr =[]
ans_arr = []

for i in range(n):
    word, num, rank = input().split()
    arr.append((int(num), word, int(rank)))

arr.sort()

for i in range(n):
    rank = arr[i][2]
    temp = arr[i][1][rank-1]
    if type(temp) is str:
        temp = temp.upper()
    ans_arr.append(temp)

print(''.join(ans_arr))